src/libraries/JANA/Services/JParameterManager.h

```diff
@@ -293,89 +307,51 @@ inline T JParameterManager::RegisterParameter(std::string name, const T default_
 
 /// @brief Logic for parsing different types in a generic way
 template <typename T>
-inline T JParameterManager::Parse(const std::string& value) {
+void JParameterManager::Parse(const std::string& value, T& val) {
     std::stringstream ss(value);
-    T val;
     ss >> val;
-    return val;
 }
 
-/// @brief Specialization for handling strings that don't need parsing
+
+/// @brief Specialization for string. 
+/// The stream operator is not only redundant here, but it also splits the string (see Issue #191)
 template <>
-inline std::string JParameterManager::Parse(const std::string& value) {
-    return std::string(value);
+inline void JParameterManager::Parse(const std::string& value, std::string& out) {
+    out = value;
 }
 
+
 /// @brief Specialization for bool
 template <>
-inline bool JParameterManager::Parse(const std::string& value) {
-    if (value == "0") return false;
-    if (value == "1") return true;
-    if (value == "false") return false;
-    if (value == "true") return true;
-    if (value == "off") return false;
-    if (value == "on") return true;
-    throw JException("'%s' not parseable as bool", value.c_str());
+inline void JParameterManager::Parse(const std::string& value, bool& val) {
+    if (value == "0" || value =="false" || value == "off") val = false;
+    else if (value == "1" || value == "true" || value == "on") val = true;
+    else throw JException("'%s' not parseable as bool", value.c_str());
 }
```

Comment:
It's probably fine here, but in general I'd advise against rearranging ugly logic unless it is truly necessary, particularly when parsing is involved. It's just so easy to introduce bugs or activate bugs that have been lying dormant. Particularly when our test cases are this light.